lib/mail/message.rb

```diff
@@ -1994,14 +1992,44 @@ def text?
     # Additionally, I allow for the case where someone might have put whitespace
     # on the "gap line"
     def parse_message
-      header_part, body_part = raw_source.lstrip.split(HEADER_SEPARATOR, 2)
-      self.header = header_part
-      self.body   = body_part
+      stream = StringIO.new(raw_source)
+      self.header = parse_headers(stream)
+      self.body = stream.read
+    end
+
+    # Parses the headers from the input stream,
+    # leaving the position at the start of the body.
```

Comment:
I figured it would be the best way to actually leave the body untouched since I'm only taking the stuff away I don't need from the stream. Splits and replaces might lead ruby to do weird stuff with the binary content.